enhancements/subscription-content/subscription-injection.md

```diff
@@ -0,0 +1,487 @@
+  shares:
+  - name: <name>-subscription-entitlement # shares are cluster-scoped
+  - name: <name>-subscription-yumRepository
+```
+
+```yaml
+kind: Bundle
+apiVersion: subscription.openshift.io/v1alpha1
+metadata:
+  name: <name>
+  namespace: <namespace>
+  ... # standard metadata
+spec:
+  entitlements:
+  - name: <secret name with entitlement keys>
+    namespace: <namespace containing the secret>
+  yumRepositories:
+  - name: <configMap name with yum repo definition>
+    namespace: <namespace for the yum repo definition>
+status:
+  conditions:
+  - type: Invalid
+    status: "False" # standard k8s conditions, should obey the "abnormal-true" convention
+  - ... # additional informational conditions
+```
+
+### Risks and Mitigations
+
+**Risk:** Entitlement keys can leak outside of the `openshift-config` namespace to unauthorized
+workloads.
+
+_Mitigation:_ Access to the entitlement keys is gated by the cluster role generated for the
+projected resource shares associated with the bundle. Cluster admins are responsible for either
+
+1. Creating a `ClusterRoleBinding` for the service accounts that need access to the entitlement, OR
+2. Declaring cluster roles to aggregate the `ClusterBundle` to via the `aggregateToClusterRoles`
+   array, OR
+3. Configuring the operator to aggregate the generated `ClusterRole` to appropriate system roles by
+   default.
+
+**Risk:** Shares and cluster roles are orphaned if the subscription `ClusterBundle` is deleted.
+
+_Mitigation:_ Owner references are used to ensure proper cleanup. Note that `ClusterBundle` is
+cluster-scoped, so it can an create owner references on any object.
+
+## Design Details
+
+### Test Plan
+
+This feature set can be tested as follows:
+
+1. Download an OpenShift entitlement key set from the customer portal, and add it to the
+   `openshift-config` namespace as the `etc-pki-entitlement` `Secret`.
+2. Create a `ClusterBundle` object 
+2. Run a pod with the `"subscription.openshift.io/inject-cluster-bundle": cluster` annotation, and
+   a UBI8 container which performs a `yum install` of a RHEL package that requires a subscription.
+
+### Graduation Criteria
+
+This feature will likely require Dev Preview, Tech Preview, and GA maturity levels.
+
+#### Feature Roadmap
+
+1. `Bundle` supporting a single bundle injected into a workload within the same namespace, allowing
+   one entitlement secret and one yum.repo configuration.
+2. `Bundle` supporting multiple `entitlements` and `yumRepositories`.
+3. Allow multiple `Bundles` to be injected into a workload within the same namespace.
+4. `ClusterBundle` supporting a single bundle injected into a workload, allowing one entitlement
+   secret and one yum.repo configuration.
+4. `ClusterBundle` supporting multiple multiple `entitlements` and `yumRepositories`.
+5. Allow multiple `ClusterBundles` to be injected into a workload.
+6. Allow auto-injection of `ClusterBundles` into all workloads. 
```

Comment:
any intent to make `Bundles` be autoinjectable also?  (within their namespace of course)

Reply:
Yeah...we should probably have that feature, too.